### DISCUSSION TITLE
PKL Support for code syntax highlighting and navigation

### DISCUSSION THREAD

**Discussion by BlueFalconHD** [community/community#107624] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Apple recently introduces Pkl, a configuration programming langauge.

It would be great if GitHub added support for Pkl syntax highlighting. The Pkl treesitter thing is here: https://github.com/apple/tree-sitter-pkl.

**Comment by dcreager** [comment]
First step would be updating the language detection rules in [linuist](/github-linguist/linguist) to detect Pkl files.  Once that is done we can use the tree-sitter grammar that you linked for syntax highlighting.  You'd also need to add some [tagging rules](https://tree-sitter.github.io/tree-sitter/code-navigation-systems) to the grammar to enable search-based Code Navigation.

**Comment by alexfalkowski** [comment]
I found this https://github.com/github-linguist/linguist/pull/6730
